lib/type-checker.js

```diff
@@ -398,10 +393,20 @@ class TypeChecker {
   }
 
   _stringTypes(string, highPrecisionSupport) {
+    let types = [ 'String', 'Object', 'Array' ];
```

Comment:
Discussed privately that as this repo is already requiring ES6, and these are easy to change, happy to downgrade and verify via [adding node 5 back into `.travis.yml`](https://github.com/mongodb-js/hadron-type-checker/commit/b57c10a13f6222f269d2804b580ae237772a5050) if it becomes useful to someone:

<img width="485" alt="this repo is already using es6 though we can downgrade and verify via travis yml if it becomes useful to someone" src="https://cloud.githubusercontent.com/assets/1217010/24283195/7863c476-10b8-11e7-82c1-7a6dd37b204c.png">

Until that's decided, we'll take whatever each repo is currently doing.